src/webnn_native/ops/Concat.cpp

```diff
@@ -17,6 +17,30 @@
 #include "webnn_native/Error.h"
 
 namespace webnn_native { namespace op {
+    MaybeError Concat::CalculateShape() {
+        auto inputShape =
+            mInputs[0]->Shape().empty() ? std::vector<int32_t>{1} : mInputs[0]->Shape();
+        int32_t axisShape = 0;
+        for (auto& input : mInputs) {
+            auto shape = input->Shape();
+            if (shape.size() != inputShape.size()) {
+                return DAWN_VALIDATION_ERROR("The input shapes are incompatible.");
```

Comment:
The error message would be better to be specific, like "The input tensors must have the same rank."